### PR TITLE
Add lobbies

### PR DESCRIPTION
**Pull request by jonty-kennedy-trendsales**
Added endpoints (I don't know if we need all of these yet):
* `POST /lobbies` -> Creates a public or private lobby associated with a host user and returns a lobby_code that other users can use to join the lobby
* `POST /lobbies/<lobby_code>/join` -> Takes a lobby code and assigns the user to that lobby
* `GET /lobbies` -> Returns all **public** lobbies
* `GET /lobbies/<lobby_code>` -> Returns details about a specific lobby
* `PUT /lobbies/<lobby_code>/users/<user_id>` -> Update details about a user (right now `username` is the only thing that can be changed)

https://github.com/user-attachments/assets/acba7a44-f7d6-49ea-a498-0a31787a6000

### CODE REVIEW

File: backend/lobby/lobby_controller.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+def get_all_public_lobbies() -> Tuple[List[Lobby], int]:
+    public_lobbies = [
+        {code: lobby}
+        for code, lobby in lobbies.items()
+        if not lobby['private']
+    ]
+
+    return jsonify(public_lobbies), 200
+
+# Technically anyone can change anyones username unless we add authentication
```
Comment (VioletteDrch) [inline backend/lobby/lobby_controller.py:82]: in later iterations 👍

**Review by VioletteDrch — COMMENTED** [review]
dope. so this is like the waiting room the players join before starting the game right ?

File: backend/lobby/lobby_controller.py (hunk 1 above)
Comment (jonty-kennedy-trendsales) [reply backend/lobby/lobby_controller.py:82]: Yeah they join this lobby before the game starts to wait for everyone to join. The plan is once a user has joined successfully, they make the web socket connection to get updates on new users joining the lobby, when the game starts, etc. We can also add like a "mark as ready" button and a limit on players in a lobby

Comment (VioletteDrch) [reply backend/lobby/lobby_controller.py:82]: Great, do you know the app PhotoRoulette ? That is kinda what I have in mind. We see the users usernames when they enter the lobby, and the host sees a button 'start the game', while others have 'waiting for the host to start the game' or smth like that.

Comment (jonty-kennedy-trendsales) [reply backend/lobby/lobby_controller.py:82]: So the host can start it at any time or everyone has to mark themself as ready first?

**Pull request merged**